test/lib/apq/document_provider_test.exs

```diff
@@ -94,6 +94,27 @@ defmodule Apq.DocumentProviderTest do
     assert resp_body == @result
   end
 
+  test "decodes 'extensions' params if sent as JSON" do
+    digest = sha256_hexdigest(@query)
+    query = @query
+    extensions = Jason.encode!(%{"persistedQuery" => %{"version" => 1, "sha256Hash" => digest}})
+
+    Apq.CacheMock
+    |> expect(:put, fn ^digest, ^query -> {:ok, query} end)
+
+    assert %{status: 200, resp_body: resp_body} =
+             conn(:post, "/", %{
```

Comment:
Why does the test use `conn(:post,...)` if the pr adds GET support?

Reply:
Because I'm absent minded 😉 

Test updated!